R/get_acoustic_detections.R

```diff
@@ -199,7 +199,7 @@ get_acoustic_detections <- function(connection = con,
   # Check limit
   assertthat::assert_that(is.logical(limit), msg = "limit must be a logical: TRUE/FALSE.")
   if (limit) {
-    limit_query <- glue::glue_sql("LIMIT 100", .con = connection)
+    limit_query <- glue::glue_sql("ORDER BY det.id_pk LIMIT 100", .con = connection)
```

Comment:
Ordering is necessary to ensure the result of the filter is always the same, if we leave out the `ORDER BY`, then a test will fail. However, this step is very expensive, defeating the point of the `LIMIT`. 

Consider restoring to unordered limiting, but documenting the behavior and perhaps rewriting the test.

Reply:
Ordering makes the examples and testing stages too slow:

```md
✔  checking examples (40m 42.8s)
   Examples with CPU or elapsed time > 5s
                             user system  elapsed
   get_animals             12.828  0.092   15.457
   get_acoustic_detections 10.664  0.552 2406.708
   get_tags                 6.940  0.040    8.048
```

Reply:
Limiting actually seems to be a little bit slower than not limiting, at least in some circumstances: 


```md

Unit: seconds
                                                                              expr
  get_acoustic_detections(con, acoustic_project_code = "demer",      limit = TRUE)
 get_acoustic_detections(con, acoustic_project_code = "demer",      limit = FALSE)
      min       lq     mean   median       uq      max neval
 151.7327 154.1687 164.0103 156.6048 170.1491 183.6935     3
 110.5260 112.2894 123.5281 114.0527 130.0292 146.0056     3
```